verifier/verifier.py

```diff
@@ -75,32 +75,29 @@ def pre_verify(self, source, rendition):
                     os.remove(audio_file)
 
             rendition_capture = cv2.VideoCapture(video_file)
-            fps = int(rendition_capture.get(cv2.CAP_PROP_FPS))
+            fps = rendition_capture.get(cv2.CAP_PROP_FPS)
             frame_count = int(rendition_capture.get(cv2.CAP_PROP_FRAME_COUNT))
             height = float(rendition_capture.get(cv2.CAP_PROP_FRAME_HEIGHT))
             width = float(rendition_capture.get(cv2.CAP_PROP_FRAME_WIDTH))
 
-            rendition_copy = rendition.copy()
             rendition['path'] = video_file
 
             # Create dictionary with passed / failed verification parameters
+            if rendition.get('resolution'):
+                rendition['resolution']['height_pre_verification'] = height / float(rendition['resolution']['height'])
+                rendition['resolution']['width_pre_verification'] = width / float(rendition['resolution']['width'])
 
-            for key in rendition_copy:
-                if key == 'resolution':
-                    rendition['resolution']['height_pre_verification'] = height / float(rendition['resolution']['height'])
-                    rendition['resolution']['width_pre_verification'] = width / float(rendition['resolution']['width'])
+            if rendition.get('frame_rate'):
+                rendition['frame_rate'] = bool(np.isclose(float(rendition['frame_rate']), fps, atol=0.1))
```

Comment:
If `rendition['frame_rate']` is 0, then I think this would evaluate to false when the actual rendition FPS is non-zero (as returned on L78). I think the behavior we want is to compare the actual rendition FPS with the source FPS when `rendition['frame_rate']` is 0.

Reply:
- It won't evaluate the condition for 0, because it will be converted to False in the if.
- The pre-verification step operate on rendition itself, if we don't have acceptable value in the metadata field, that's nothing to compare rendition fps with.

Reply:
> The pre-verification step operate on rendition itself, if we don't have acceptable value in the metadata field, that's nothing to compare rendition fps with.

My initial thoughts were:

- When `rendition['frame_rate']` is 0 that means that the broadcaster is telling the transcoder to *not* adjust the FPS of the source video (this is a current property of the Livepeer system). So, the rendition FPS should be the same as the source FPS
- The broadcaster currently does not know what the source FPS is. Otherwise, the broadcaster could set `rendition['frame_rate']` to the source FPS when it sends a request to the verifier
- The broadcaster implementation *might* end up calculating the source FPS in the future, but for now it might be ok for the verifier to calculate the source FPS in this case. Ex. When `rendition['frame_rate']` is 0, the verifier understands that the rendition should have the same FPS as the source so it:
    - Calculates the source FPS
    - Calculates the rendition FPS
    - Compares the calculated source FPS with the calculated rendition FPS
- If the broadcaster implementation ends up calculating the source FPS in the future such that it can properly populate `rendition['frame_rate']` in the request to the verifier then we can remove the above logic

With all that being said, I'm curious if calculating the source FPS in this situation would introduce additional overhead or maybe its negligible?

Reply:
Got it, frame_rate=0 means 'same as master'. Fixed the logic. The overhead is negligible as it only requires to read the video file header.